Adds the ability to apply pressure to bleeding wounds, splint rewrite, rigsuit fixes and refactor

mwerezak opened the pull request:
Adds applying pressure to body parts to reduce bleeding.
Help-intent click self with an empty hand to apply pressure to your own organs, help-intent click on someone with an aggressive grab to apply pressure to someone else's organs.

Each person should only be able to apply pressure to one body part on one person at a time, though I'm not terribly happy with the implementation. I wanted to avoid putting an extra var on humans though, as it is there is already an extra var on external organs, needed for bloodying hands (otherwise I would have used a status bitflag).

Raptor1628 commented:
How do you examine yourself now if help intent clicking applies pressure?

mwerezak commented:
You click a second time.

Asanadas commented:
Is there any room with these changes for a tourniquet item? Or, could that functionality be added to the splint medical object in some fashion? That would be a nifty extension of this feature.

comma commented:
Why up the puncture wound threshold?
They do not merge like cuts, so you'd only get big wounds from single big damages.

mwerezak commented:
5 is really low - makes sense for cuts and gashes but I figured puncture wounds are smaller in surface area and would clot more easily.

Yeah, I guess with puncture wounds unable to widen smaller sharp but not-edged melee weapons (do we have any of those?) will be unable to cause the kind of profuse bleeding that edged weapons would.

mwerezak commented:
I could make splints auto-apply pressure. I wonder if auto-splinting hardsuits should do too - that seems like it would be a huge buff since those suits splint any organ.

Tourniquet items - would have to sort out creation, sprites. I'm not sure if tourniquets really have a place in this game since it's hard to get more effective than bandages. I suppose we could always just nerf the ability of bandages to handle humongous wounds to *make* a place for tourniquets, but it's really not a priority for myself.

Asanadas commented:
Gotcha. Bandages are already super tourniquets as it is.

Techhead0 commented:
Regarding hardsuits: Users have a very hard time obtaining medical treatment if they pass out, and often aren't able to obtain medical treatment to them at all. Not bleeding out so fast seems like an okay tradeoff.

Regarding tourniquets: How about the ability to tie two rags together for a single-use bandage? Then you can tear up bedsheets in an emergency. Otherwise: 

> Bandages are already super tourniquets as it is.

Raptor1628 commented:
Yeah hardsuits are still fairly rare as well, auto-bleeding reduction makes sense.

mwerezak commented:
Had to re-rewrite how splinting is handled, managed to fix #10591 in the process though.

mwerezak commented:
Rigsuit splinting on going online/offline isn't quite working yet, update coming soon.

mwerezak commented:
Refactored how rigsuits handled going online/offline. Rigsuits now count as offline during the sealing/unsealing process. 

Updated rigsuit vision restriction to use the tint var and tint defines. Updated the existing rigsuits, also did some tweaking while I was at it - gave industrial hardsuits some online vision restriction, made combat hardsuits less OP.

mwerezak commented:
Turns out it was also possible to seal rigsuits that were completely out of power, too. That's been fixed.
